apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerClientCreationOptions.cs

```diff
@@ -76,12 +68,15 @@ private static async Task<ChannelCredentials> CreatedScopedDefaultCredentials()
 
         internal SpannerClientCreationOptions(SpannerConnectionStringBuilder builder)
         {
-            EmulatorDetection = builder.EmulatorDetection;
-            // If the client connects to the emulator use its endpoint.
-            var emulatorBuilder = MaybeCreateEmulatorClientBuilder();
+            var emulatorBuilder = new SpannerClientBuilder { EmulatorDetection = builder.EmulatorDetection }
+                .MaybeCreateEmulatorClientBuilder();
+            UsesEmulator = emulatorBuilder is object;
+            // If the client connects to the emulator use its endpoint (regardless of builder.Endpoint)
             Endpoint = emulatorBuilder?.Endpoint ?? builder.EndPoint;
             _credentialsFile = builder.CredentialFile;
```

Comment:
nit: maybe consider also using the one from the emulatorBuilder if it's not null, if at some point the emulator accepts a dummy credential we don't need to change anything here.

Reply:
I'd expect the emulator builder to load the credentials file if one has been specified. Basically it should set the endpoint and the credentials, and that's all.